Services/OperationService/Bots/BotRules/BotRule.cs

```diff
@@ -0,0 +1,132 @@
+            ICommand<GetUserBalanceRequest, UserBalance> balanceCommand,
+            ICommand<GetInstrumentsRequest, IEnumerable<Instrument>> instrumentsCommand)
+        {
+            triggers = new List<Trigger>();
+            this.tradeCommand = tradeCommand;
+            this.candlesCommand = candlesCommand;
+            this.balanceCommand = balanceCommand;
+            this.instrumentsCommand = instrumentsCommand;
+            token = request.Token;
+            userId = request.UserId;
+            timeMarker = request.TimeMarker;
+            triggerValue = request.TriggerValue;
+            operationType = request.OperationType;
+            maxBalancePercents = request.MoneyLimitPercents;
+        }
+
+        private async Task Execute(object sender, TriggerEventArgs e)
+        {
+            var userBalance = await balanceCommand.Execute(
+                new GetUserBalanceRequest()
+                {
+                    UserId = userId
+                });
+
+            var balance = e.Currency switch
+            {
+                Currency.Rub => userBalance.BalanceInRub,
+                Currency.Eur => userBalance.BalanceInEur,
+                Currency.Usd => userBalance.BalanceInUsd,
+                _ => throw new BadRequestException("Unsupported currency")
+            };
+
+            var maxPrice = balance * maxBalancePercents / 100;
+
+            await tradeCommand.Execute(
+                new TradeRequest()
+                {
+                    UserId = userId,
+                    Broker = BrokerType.TinkoffBroker,
+                    Token = token,
+                    Operation = operationType,
+                    Figi = e.Figi,
+                    Price = e.Price,
+                    Count = (int)(maxPrice / e.Price),
+                    Currency = e.Currency
+                });
+        }
+
+        public void Start(List<string> figis)
+        {
+            var instruments = instrumentsCommand.Execute(
+                new GetInstrumentsRequest()
+                {
+                    Broker = BrokerType.TinkoffBroker,
+                    Token = token,
+                    Type = InstrumentType.Any
+                }).Result;
+
+            foreach (var figi in figis)
+            {
+                try
+                {
+                    var currency = instruments.Where(x => x.Figi == figi).Select(x => x.Currency).First();
+
+                    var trigger = new TimeDifferenceTrigger(
+                       figi,
+                       timeMarker,
+                       triggerValue,
+                       token,
+                       currency,
+                       candlesCommand
+                    );
+
+                    trigger.Triggered += async (s, e) =>
+                    {
+                        await Execute(s, e);
+                    };
+
+                    triggers.Add(trigger);
+                }
+                catch
```

Comment:
May be `catch(Exception)`

Reply:
> May be `catch(Exception)`

The difference between catch and catch(Exception) is that when we write just catch, it catches all types of possible excpetions even if they are not present in .net, and actually in our case it does not make any sense, but agree, mb it's better because of common styleguide :)